Implemet web socket to get the updated Reports instead of calling th…

zakriya057 opened the pull request:
Closes #353

Commit:
Implemet web socket to get the updated Reports instead of calling the api

chgad commented on api/utilities.py:
```diff
@@ -333,3 +335,41 @@ def update_last_used_on_contract(sender, instance, created=False, **kwargs):
+
+    async_to_sync(channel_layer.group_send)(
+        f'ReportsSocket_{str(instance.user.id)}',
+        {
```
Why can't we use the ReportSerializer here?

zakriya057 replied:
Bad of me! I will change it with serializer.

chgad commented on config/asgi.py:
```diff
@@ -0,0 +1,29 @@
+import os
```
since we have the asgi.py now, do we still need the wsgi.py ?

zakriya057 replied:
We don't need it anymore I will just confirm it and will remove it.

chgad commented on config/consumers.py:
```diff
@@ -0,0 +1,33 @@
+from channels.exceptions import DenyConnection
+import json
+
+class MyConsumer(AsyncWebsocketConsumer):
```
Please consider a more fitten/professional naming.

chgad commented on config/settings/common.py:
```diff
@@ -102,7 +103,19 @@
+    'default': {
+        'BACKEND': 'channels_rabbitmq.core.RabbitmqChannelLayer',
+        'CONFIG': {
+            "host": "amqp://broker_adm:broker_pass@rabbit_broker:5672/",  # URL for RabbitMQ connection
```
This will fail in production. Use `env.str('RABBITMQ_URL')` here.

zakriya057 replied:
ok sure.

chgad commented on docker/local/start.sh:
```diff
@@ -3,8 +3,11 @@
-    echo "Re-starting Django runserver_plus!"
-    python manage.py runserver_plus 0.0.0.0:8000
+    echo "Starting Uvicorn ASGI server!"
+    uvicorn config.asgi:application --host 0.0.0.0 --port 8000
```
Just to be sure: It is actually necessary for the uviconr command to be inside the `while` loop?

Commit:
1.Change the name of Consumer
2. convert data with the help of serializer in post save function
3. get the url from env in settings.py
4. change the way how the server is starting

chgad committed:
Black & Isort

chgad committed:
Poetry lock.

chgad committed:
Remove nginx service.

chgad committed:
Remove deprecated wkpdftohtml install.

chgad committed:
Make uvicorn reload.

chgad committed:
Change jwt location to subprotocols.

chgad committed:
Refactor websocket implementation.

chgad committed:
poetry lock.

chgad committed:
Black format.
